### PR TITLE
Adds shock arm

### PR DESCRIPTION
**Pull request by FriendlyContractor**
## About The Pull Request
Adds new unique weapon - shock arm. Shock arm has a range of 5 tiles and once you lock onto somebody it will continously arc to them dealing 15 stamina damage every 2 seconds until they break line of sight or move out of range. Damage is reduced by stamina armor. Oh and most importantly it replaces one of your arms. Besides adding shock arm chainsaw arm code has been somewhat improved removing repeated code and moving it to separate file
## Why It's Good For The Game
Adds new content for security expanding their non-lethal weapon options

## Changelog
:cl:
add: Added shock arm - unique weapon that locks onto target and incapiciates them with electric shocks. Can be printed in security autolathe after it has been researched. Replaces one of your arms
refactor: Slightly refactored chainsaw arm code
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
Should probably mention this is a refactor.

**Comment by RDS88-beeman** [comment]
did you account for the fact stamina regenerates like oxyloss over here instead of just hanging out until it's all cleared after 10 seconds? It feels like this is out-regenable rather easily but i don't know the numbers, just the vibe

**Comment by FriendlyContractor** [comment]
Yes. From my testing in game stamina damage applied as intended and was not out-healed

**Comment by FriendlyContractor** [comment]
Hmm there is something my code that linter doesnt like. I would appreciate some help here since im not exactly the greatest coder and have have no idea what it is ;-;

**Comment by RDS88-beeman** [comment]
also wait does this turn off when shoved? or dead?

File: code/game/objects/items/item_arms.dm [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		var/obj/item/bodypart/affecting = shock_target.get_bodypart(ran_zone(BODY_ZONE_CHEST))
+		var/armor_block = shock_target.run_armor_check(affecting, "stamina")
+		shock_target.apply_damage(shock_stamina, STAMINA, affecting, armor_block)
+		shock_target.electrocute_act(0, shock_instigator, 1, FALSE, FALSE, FALSE, FALSE, FALSE)
+		playsound(src, 'sound/weapons/zapbang.ogg', 50)
+	else
+		is_shocking = FALSE
+		STOP_PROCESSING(SSobj, src)
+
+obj/item/item_arm/shock_arm/attack_self(mob/user)
```
Comment (Sarchutar) [inline code/game/objects/items/item_arms.dm]:
```suggestion
/obj/item/item_arm/shock_arm/attack_self(mob/user)
```

File: code/game/objects/items/item_arms.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	icon_state = "shock_arm"
+	item_state = "shock_arm"
+	lefthand_file = 'icons/mob/inhands/weapons/item_arm_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/item_arm_righthand.dmi'
+	droped_object = /obj/item/shock_arm_unattached
+	var/shock_range = 5
+	var/shock_stamina = 15
+	var/mob/living/shock_target
+	var/mob/living/shock_instigator
+	var/is_shocking = TRUE
```
Comment (Sarchutar) [inline code/game/objects/items/item_arms.dm]: `is_shocking` doesn't do anything, it never gets called it only gets set to either `TRUE` or `FALSE`

**Comment by zeskorion** [comment]
Adding a new limb replacing weapon violates the carbon freeze afaik

File: code/game/objects/items/item_arms.dm (hunk 2 above)
Comment (FriendlyContractor) [reply code/game/objects/items/item_arms.dm]: Thank you for taking your time to check my code and point this out

**Comment by FriendlyContractor** [comment]
> Adding a new limb replacing weapon violates the carbon freeze afaik

It doesnt touch any limb or carbon related code, it only changes 5 lines in surgery. If it still violates carbon freeze please take my appologies and close it

**Comment by ike709** [comment]
DNM due to the code freeze on carbon mobs.

**Comment by Crossedfall** [comment]
@Kapu1178 Will need to actually quantify things from here. But I've added the DNM label for now.

**Comment by Kapu1178** [comment]
Pseudolimbs are more than likely going to be rewritten due to them being a nightmare mess. At worst you'll need to remake the PR once #5611 is merged.

At best, you won't really need to do anything, but this is more for my sanity.

**Comment by VictorVow** [comment]
How many seconds to stamcrit?

File: code/game/objects/items/item_arms.dm [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	w_class = WEIGHT_CLASS_HUGE
+	var/obj/droped_object
+
+/obj/item/item_arm/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
+
+/obj/item/item_arm/Destroy()
+	var/obj/item/bodypart/part
+	new droped_object(get_turf(src))
```
Comment (Fikou) [inline code/game/objects/items/item_arms.dm:14]: destroy is for cleaning up references, move this to a natural destruction proc and make this just QDEL_NULL the object

File: code/game/objects/items/item_arms.dm [hunk 4]
```diff
@@ -0,0 +1,111 @@
+/obj/item/item_arm
+	name = "prosthetic item template"
+	desc = "A piece of equipment that replaced your arm"
+	item_flags = ABSTRACT | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	var/obj/droped_object
```
Comment (Fikou) [inline code/game/objects/items/item_arms.dm:6]: thats not how you spell dropped

File: code/game/objects/items/item_arms.dm [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+/obj/item/shock_arm_unattached
+	name = "Unattached Shock Arm"
+	desc = "Shock arm ready to be surgicaly attached"
+	icon = 'icons/obj/item_arms.dmi'
+	icon_state = "shock_arm"
+
+/obj/item/item_arm/shock_arm
+	name = "Shock Arm"
+	desc = "Device capable of incapiciating oponents at range using low intensity electricity arcs."
+	force = 0
```
Comment (Fikou) [inline code/game/objects/items/item_arms.dm:33]: inherited from parent

File: code/game/objects/items/item_arms.dm [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		var/mob/living/carbon/holder = loc
+		var/index = holder.get_held_index_of_item(src)
+		if(index)
+			part = holder.hand_bodyparts[index]
+	. = ..()
+	if(part)
+		part.drop_limb()
+
+/obj/item/shock_arm_unattached
+	name = "Unattached Shock Arm"
```
Comment (Fikou) [inline code/game/objects/items/item_arms.dm:25]: item names are usually not capitalized

File: code/game/objects/items/item_arms.dm [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+	desc = "Shock arm ready to be surgicaly attached"
+	icon = 'icons/obj/item_arms.dmi'
+	icon_state = "shock_arm"
+
+/obj/item/item_arm/shock_arm
+	name = "Shock Arm"
```
Comment (Fikou) [inline code/game/objects/items/item_arms.dm:31]: item names are usually not capitalized

**Comment by FriendlyContractor** [comment]
Bad code and lack of will to maintain. Feel free to steal the idea

**Pull request closed without merging**